### ISSUE TITLE
Is it correct to just use visual encoder&prediction head during both training/inference?

### ISSUE THREAD

**Issue by LiQiiiii** [jzhang538/BadMerging#1]
Hi, I've read your code and find that you just use the visual encoder and a prediction head during training, and during inference you just upload all the fine-tuned visual encoder and merge them, it seems that you downgraded the problem to a pure image classification problem, i.e.,  putting aside model merge, there's no difference between your work and other backdoor works in vision domain before. In my opinion, at least in the inference, you need to use the text encoder for prediction, just same as in the Figure 1 of your paper.

**Comment by jzhang538** [comment]
Hi, 

Thanks so much for your question! I summarize your question into the following sub-questions and provide an answer to each of them:

---

**(1) Why do you use a fixed prediction head during both training and inference?** 

We strictly follow the pipeline of existing works (e.g., Task Arithmetic) in model merging, where the text encoder is frozen during both fine-tuning and merging processes. Since the text encoder is frozen, both the pre-trained model and the merged model will essentially share the same prediction head for a specific task. The variation only stems from the random seed used during text feature extraction. To this end, we can simply construct a prediction head using the initial text encoder and keep using this prediction head for both training and inference (This is also how existing works did). Please also refer to their codebase: 

(a) Task Arithmetic: https://github.com/mlfoundations/task_vectors/tree/main/src
(b) AdaMerging: https://github.com/EnnengYang/AdaMerging/tree/main/src

Actually, we also tried to re-run the text encoder to construct a new prediction head for each specific task during inference. The results won't be affected due to the minor difference between prediction heads.

**(2) What's the difference between BadMerging and traditional backdoor attacks?** 

For traditional backdoor attacks, their embedded backdoor may be compromised during the (unknown) model merging process. As shown in Table 2-3, all the existing backdoor attacks fail in the context of model merging. In contrast, BadMerging comprises a two-stage attack mechanism and a novel feature-interpolation-based loss to enhance the robustness of embedded backdoors against the changes of merging parameters.

---
Please feel free to let me know if you have further questions!

**Comment by LiQiiiii** [comment]
Hi, thanks for your reply. From this perspective, this job seems to have nothing to do with whether it is a CLIP or not. It is essentially a backdoor for ViT, right?

**Comment by jzhang538** [comment]
Thanks for the question!

**Our backdoor attack is not tailored to ViT. We highlight that the main focus of this work is how to make the backdoor survive from the (unknown) model merging process.** Existing works on model merging focus on merging the visual encoder (ViT) of CLIP-like models and we follow the same setting. However, the key idea can be extended to model merging for other model designs (e.g., LLM).

**Comment by LiQiiiii** [comment]
Thanks for your reply. Let me put it in this way: It is purely a vision task and has nothing to do with text, right?

**Comment by LiQiiiii** [comment]
> Thanks for your reply. Let me put it in this way: It is purely a vision task and has nothing to do with text, right?

At least in your current work, despite the potential for scaling up to VLM/MLLM or something.

**Comment by jzhang538** [comment]
For this code implementation, yes!!

**Comment by LiQiiiii** [comment]
I see I see, thanks for the patience.
